src/testWithSpringBoot_2_7/java/org/openrewrite/java/spring/boot2/UpdateMysqlDriverArtifactIdTest.java

```diff
@@ -106,34 +105,103 @@ void doNotPinWhenNotVersioned() {
-                      </dependency>
-                    </dependencies>
-                  </project>
-                  """, version.group(0));
-            })
-          )
-        );
+              """, spec -> spec.after(pom -> {
+                  Matcher version = Pattern.compile("2.7.\\d+").matcher(pom);
+                  assertThat(version.find()).describedAs("Expected 2.7.x in %s", pom).isTrue();
+                  //language=xml
+                  return String.format("""
+                                       <project>
+                                         <modelVersion>4.0.0</modelVersion>
+                                         <groupId>com.example</groupId>
+                                         <artifactId>demo</artifactId>
+                                         <version>0.0.1-SNAPSHOT</version>
+                                         <parent>
+                                           <groupId>org.springframework.boot</groupId>
+                                           <artifactId>spring-boot-starter-parent</artifactId>
+                                           <version>%s</version>
+                                           <relativePath/> <!-- lookup parent from repository -->
+                                         </parent>
+                                         <dependencies>
+                                           <dependency>
+                                             <groupId>com.mysql</groupId>
+                                             <artifactId>mysql-connector-j</artifactId>
+                                             <scope>runtime</scope>
+                                           </dependency>
+                                         </dependencies>
+                                       </project>
+                                       """, version.group(0));
+              })));
+        }
+    }
+
+    @Nested
+    class Gradle {
+
+        @DocumentExample
+        @Test
+        void gradleSwitchArtifactIdAndUpdateVersionNumber() {
+            rewriteRun(spec -> spec.beforeRecipe(withToolingApi()),
+                       //language=groovy
+                       Assertions.buildGradle("""
+                                              plugins {
+                                                id 'java'
+                                              }
+                                                                   
+                                              repositories {
+                                                 mavenCentral()
+                                              }
+                                                                   
+                                              dependencies {
+                                                  runtimeOnly 'mysql:mysql-connector-java:8.0.30'
+                                              }
+                                              """, """
```

Comment:
stylistic, put another text block to a new line.

Reply:
Moved text blocks to new lines